### ISSUE TITLE
Error: Pandoc Export Failed

### ISSUE THREAD

**Issue by abrahambahez** [OliverBalfour/obsidian-pandoc#50]
OS: ElementaryOS Hera (based on Ubuntu 18.04.4 LTS)

Issue: I cannot get working the plugin in Linux although I use pandoc CLI very often without any problem. The output of `which pandoc` command inside the terminal is right: `/usr/bin/pandoc`. I've install pandoc from a .deb package and I'm wondering if it's because some path divergence from the installation method; but I'm lost. I attached the Obsidian console output of the errors for every path I have find a pandoc file.

Any idea of what is going wrong?

![Screenshot from 2021-07-13 23-45-03](https://user-images.githubusercontent.com/53906659/125562923-0e4cc3ec-eb73-40c4-9afa-a69e01a30751.png)

**Comment by OliverBalfour** [comment]
Hmm, that's odd! Works fine on my Linux machine. What happens if you manually set the `Pandoc path` setting to `/usr/bin/pandoc`?

**Comment by Leyeev** [comment]
Got the same problem I'm using gnome 40, It tried both` /usr/bin/pandoc `and `which pandoc`.

**Comment by Carriolan** [comment]
OS: Mac OS 14.1.2 (16611.3.10.1.3) Obsidian 0.12.12. Pandoc Plugin v0.2.5 Pandoc Installer pandoc-2.14.1-macOS
I have the same error "Pandoc Export Failed:" Attached are screenshots of my parameters.

Installation:
1. I used the standard Pandoc installer package (see above for version)
2. I checked that the relevant .obsidian folder had all of the requisite files
3. In parameters I referenced the style.css folder.

I am stumped!
<img width="1153" alt="Screenshot 2021-08-06 at 12 26 57" src="https://user-images.githubusercontent.com/87270337/128504620-4912c5f2-78a7-4740-a651-426e9c913067.png">
<img width="1156" alt="Screenshot 2021-08-06 at 12 26 27" src="https://user-images.githubusercontent.com/87270337/128504631-99c2004c-9732-44aa-bedd-9160de8d718d.png">

**Comment by Davidsoff** [comment]
@abrahambahez Looks like the plugin does not work if you have Obsidian installed using snap.
Installing using deb and then using the aut upgrade solved the issue for me

**Comment by abrahambahez** [comment]
> @abrahambahez Looks like the plugin does not work if you have Obsidian installed using snap.
> Installing using deb and then using the aut upgrade solved the issue for me

Yes, indeed! It seems to work with the AppImage installation too, but not with the flatpak nor the snap ones

**Comment by abrahambahez** [comment]
> OS: Mac OS 14.1.2 (16611.3.10.1.3) Obsidian 0.12.12. Pandoc Plugin v0.2.5 Pandoc Installer pandoc-2.14.1-macOS
> I have the same error "Pandoc Export Failed:" Attached are screenshots of my parameters.
> 
> Installation:
> 
> 1. I used the standard Pandoc installer package (see above for version)
> 2. I checked that the relevant .obsidian folder had all of the requisite files
> 3. In parameters I referenced the style.css folder.
> 
> I am stumped!
> <img alt="Screenshot 2021-08-06 at 12 26 57" width="1153" src="https://user-images.githubusercontent.com/87270337/128504620-4912c5f2-78a7-4740-a651-426e9c913067.png">
> <img alt="Screenshot 2021-08-06 at 12 26 27" width="1156" src="https://user-images.githubusercontent.com/87270337/128504631-99c2004c-9732-44aa-bedd-9160de8d718d.png">

In your case I see that you put the literal words 'which pandoc' but you shuld put the output of that command in a terminal.

You could try to:

1. Leave the "Pandoc path" field empty to use the default, or
2. Open a terminal in your Mac, write 'which pandoc' in it, press enter, copy the output and paste it in the "Pandoc path" field

I think this should solve that issue

**Comment by Davidsoff** [comment]
why is the issue closed? the fact that the plugin fails for snap and flatpack installs is still there?

**Comment by Carriolan** [comment]
Thanks David - It worked!

--
Peter Sheffield
